### PR TITLE
enable running smoke tests in staging repo

### PR DESCRIPTION
**Pull request by neozwu**
This change is to enable running smoke tests in staging repo (by invoking "./gradlew smokeTest").

### CODE REVIEW

File: src/main/resources/com/google/api/codegen/java/build_gapic.gradle.snip [hunk 1]
```diff
@@ -21,7 +21,9 @@
     compile 'com.google.api:gax:{@metadata.gaxVersionBound.lower}'
     compile 'com.google.api:gax-grpc:{@metadata.gaxGrpcVersionBound.lower}'
     testCompile 'junit:junit:4.12'
+    testCompile 'commons-lang:commons-lang:2.6'
```
Comment (garrettjonesgoogle) [inline src/main/resources/com/google/api/codegen/java/build_gapic.gradle.snip]: Shouldn't this be removed?

File: src/main/resources/com/google/api/codegen/java/build_gapic.gradle.snip [hunk 2]
```diff
@@ -21,7 +21,9 @@
...
     testCompile 'com.google.api:gax-grpc:{@metadata.gaxGrpcVersionBound.lower}:testlib'
+    testCompile 'io.grpc:grpc-netty-shaded:{@metadata.grpcVersionBound.lower}'
```
Comment (garrettjonesgoogle) [inline src/main/resources/com/google/api/codegen/java/build_gapic.gradle.snip]: Why is this necessary?

File: src/main/resources/com/google/api/codegen/java/build_gapic.gradle.snip (hunk 1 above)
Comment (neozwu) [reply src/main/resources/com/google/api/codegen/java/build_gapic.gradle.snip]: Done. Yes, I guess it's just really hard to actually remove commons-lang.

File: src/main/resources/com/google/api/codegen/java/build_gapic.gradle.snip (hunk 2 above)
Comment (neozwu) [reply src/main/resources/com/google/api/codegen/java/build_gapic.gradle.snip]: I think we need this or the right combination of  (grpc-netty + netty-tcnative-boringssl-static) to send real requests. I figured using grpc-netty-shaded is a lot easier to manage. But you probably have more insights here.

**Comment by neozwu** [comment]
@garrettjonesgoogle PTAL

File: src/main/resources/com/google/api/codegen/java/smoke_test.snip [hunk 3]
```diff
@@ -11,6 +11,11 @@
     private static final String PROJECT_ENV_NAME = "GOOGLE_CLOUD_PROJECT";
     private static final String LEGACY_PROJECT_ENV_NAME = "GCLOUD_PROJECT";
 
+    @@Test
+    public void run() {
+      main(null);
+    }
```
Comment (garrettjonesgoogle) [inline src/main/resources/com/google/api/codegen/java/smoke_test.snip]: Will smoke tests run when we do `./gradlew test`? It seems like that would make api-client-staging fail, right? (since a project wouldn't be defined)

Comment (neozwu) [reply src/main/resources/com/google/api/codegen/java/smoke_test.snip]: It will. And it will fail unless you have an env var to denote project_id.  I now configured the build script so that smoke tests are excluded when running `./gradlew test`.
